moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/parameterization/model_based_state_space.h

```diff
@@ -86,84 +87,148 @@ class ModelBasedStateSpace : public ompl::base::StateSpace
       IS_GOAL_STATE = 16
     };
 
-    StateType() : ompl::base::State(), values(nullptr), tag(-1), flags(0), distance(0.0)
+    struct alignas(16) AtomicCache
```

Comment:
I think adding some of the explanation from the PR here as a docstring is useful. It could contain:
- Why this struct exists
- Why the `alignas(16)` is there (or a link to an external resource that explains this).

Having a docstring there compared to having to dig up this PR is maybe easier for future readers?